src/main/kotlin/ecommerce/api/ProductController.kt

```diff
@@ -0,0 +1,57 @@
+package ecommerce.api
+
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import java.net.URI
+
+@RestController
+@RequestMapping("/api")
+class ProductController(private val productService: ProductService) {
```

Comment:
You're using `Constructor Injection` for DI 👍 
Have you heard of `Field Injection` and `Setter Injection` as well? Do you know how they differ?